### PR TITLE
Nextjs: Fix v13.5.x

### LINKED ISSUE #24364: [Bug]: Storybook does not work with "next": "13.5.2-canary.0"
### Describe the bug

After upgrading to Next v13.5.2-canary.0, @storybook/nextjs fails to start.
Error I am facing. 
```
ModuleNotFoundError: Module not found: Error: Can't resolve 'n
ext/dist/shared/lib/router-context' in 'D:\Users\adarsh.raj\De
sktop\ajnaai\ajna-ai\node_modules\@storybook\nextjs\dist'
```
https://github.com/storybookjs/storybook/issues/24234 This issue fixes the problem with only 13.5.1 after i updated nextjs. It came up again

### To Reproduce

To Reproduce
Upgrade Next to latest v13.5.2-canary.0. Install latest of @storybook/nextjs,  "^7.4.2"

### System

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by valentinpalkovic**
Closes https://github.com/storybookjs/storybook/issues/24364, https://github.com/storybookjs/storybook/issues/24304

## What I did

Readded Angular 17 prerelease sandbox and added Next.js canary

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

File: scripts/sandbox/generate.ts [hunk 1]
```diff
@@ -235,6 +235,7 @@ export const generate = async ({
   debug,
 }: OptionValues<typeof options>) => {
   const generatorConfigs = Object.entries(sandboxTemplates)
+    .filter(([key]) => key === 'nextjs/prerelease' || key === 'angular-cli/prerelease')
```
Comment (JReinhold) [inline scripts/sandbox/generate.ts]: Is this temporary? Looks like we're only generating the two sandboxes now.

**Comment by valentinpalkovic** [comment]
Sorry @JReinhold. This PR is still WIP. Converted it to draft

File: .github/workflows/generate-sandboxes-next.yml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
         with:
-          ref: next
+          ref: valentin/setup-sandboxes
```
Comment (yannbf) [inline .github/workflows/generate-sandboxes-next.yml:24]: reminder to bring this back to next

Comment (yannbf) [inline .github/workflows/generate-sandboxes-next.yml:24]: reminder to bring this back to next
```suggestion
          ref: next
```

File: code/frameworks/nextjs/src/dependency-map.ts [hunk 3]
```diff
@@ -6,17 +6,17 @@ const mapping: Record<string, Record<string, string>> = {
...
   },
-  '>=13.5.0': {
-    'next/dist/shared/lib/router-context': 'next/dist/shared/lib/router-context.shared-runtime',
-    'next/dist/shared/lib/head-manager-context':
-      'next/dist/shared/lib/head-manager-context.shared-runtime',
-    'next/dist/shared/lib/app-router-context':
-      'next/dist/shared/lib/app-router-context.shared-runtime',
-    'next/dist/shared/lib/hooks-client-context':
-      'next/dist/shared/lib/hooks-client-context.shared-runtime',
+  '< 13.5.0': {
```
Comment (yannbf) [inline code/frameworks/nextjs/src/dependency-map.ts:12]: Will this ever be hit? given that the `>=11.1.0 <13.5.0` also includes `<13.5.0`. Also, there's a space there, I don't know if it matters.

```suggestion
  '<13.5.0': {
```

**Review by yannbf — APPROVED** [review]
LGTM! Thanks for doing this

**Pull request merged**